packages/smooth_app/lib/pages/product/edit_ingredients_page.dart

```diff
@@ -120,44 +117,40 @@ class _EditOcrPageState extends State<EditOcrPage> {
     }
   }
 
-  Future<bool> _updateText(final String text, UpToDateProductProvider provider,
-      ImageField imageField) async {
+  Future<bool> _updateText(
+    final String text,
+    final ImageField imageField,
+  ) async {
     final LocalDatabase localDatabase = context.read<LocalDatabase>();
-    final DaoProduct daoProduct = DaoProduct(localDatabase);
-    Product changedProduct = Product(barcode: _product.barcode);
-    Product? cachedProduct = await daoProduct.get(_product.barcode!);
-    if (cachedProduct != null) {
-      cachedProduct = _helper.getMinimalistProduct(cachedProduct, text);
-    }
-    changedProduct = _helper.getMinimalistProduct(changedProduct, text);
-    await BackgroundTaskDetails.addTask(
-      changedProduct,
+    final Product minimalistProduct = _helper.getMinimalistProduct(
+      Product(barcode: _product.barcode),
+      text,
+    );
+    if (!await BackgroundTaskDetails.addTask(
+      localDatabase: localDatabase,
+      minimalistProduct: minimalistProduct,
       productEditTask:
           _helper.getImageField().value == ImageField.PACKAGING.value
               ? ProductEditTask.packaging
               : ProductEditTask.ingredient,
-    );
-    final Product upToDateProduct = cachedProduct ?? changedProduct;
-    await daoProduct.put(upToDateProduct);
-    provider.set(upToDateProduct);
-    localDatabase.notifyListeners();
-    if (!mounted) {
+    )) {
       return false;
     }
-    ScaffoldMessenger.of(context).showSnackBar(
-      SnackBar(
-        content: Text(
-          AppLocalizations.of(context).product_task_background_schedule,
-        ),
-        duration: SnackBarDuration.medium,
-      ),
+    if (!mounted) {
+      return true;
+    }
+    AbstractBackgroundTask.showSnackBar(
```

Comment:
Kind of weird to mix a task (functional part) vs UI code

Reply:
I'm not sure I see what you mean.
From the UI the user clicked: a background task is created and we display a "a background task was created" message.
Again, this was not an invention from this particular PR.